Assets/Scripts/CharacterControllers/Melody/States/AttackState.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Melody.States
+{
+    public class AttackState : MelodyState
+    {
+        public AttackState(MelodyController controller) : base(controller) { }
+
+        protected override void Enter()
+        {
+            melodyController.animator.SetTrigger("Attack");
+        }
+
+        public override void OnUpdate(float time)
+        {
+            base.OnUpdate(time);
+            if (melodyController.animator.IsInTransition(0) && melodyController.animator.GetCurrentAnimatorStateInfo(0).IsName("Attack"))
+            {
+                new IdleState(melodyController);
```

Comment:
What is the purpose of this line if it isn't stored anywhere, and it just gets called again on the very next line?